shell/imports/client/grain-client.js

```diff
@@ -1528,6 +1528,14 @@ Meteor.startup(function () {
       // TODO(security): defend against malicious apps spamming this call, blocking all other UI.
       const currentGrain = globalGrains.getActive();
       if (senderGrain === currentGrain && !globalTopbar.isPopupOpen()) {
+        try {
+          check(event.data.startSharing, {hash: Match.Maybe(String), pathname: Match.Maybe(String)});
+        } catch (err) {
+          console.error(err);
+          console.log("startSharing data is not the expected shape.");
+          console.log("See https://docs.sandstorm.io/en/latest/developing/path/#helping-the-user-share-access");
```

Comment:
Nitpick: these should probably also be console.error

Reply:
The console.log looked better in the console IMO.  But nitpicks must be smashed.